test/suite/test_compact07.py

```diff
@@ -0,0 +1,212 @@
+            files_compacted = 0
+            for i in range(self.n_tables):
+                uri = self.uri_prefix + f'_{i}'
+                if (self.get_pages_rewritten(uri) > 0):
+                    files_compacted += 1
+                    
+            time.sleep(5)
+            
+        # Check that we made no progress on the small file.
+        self.assertEqual(self.get_pages_rewritten(self.uri_prefix + '_small'), 0)
+        
+        # Check the background compaction server stats. We should have skipped at least once and 
+        # been successful at least once.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        skipped = stat_cursor[stat.conn.background_compact_skipped][2]
+        success = stat_cursor[stat.conn.background_compact_success][2]
+        self.assertGreater(skipped, 0)
+        self.assertGreater(success, 0)
+        stat_cursor.close()
+        
+        # Open a new session for foreground compaction, while leaving the background
+        # compaction server running.
+        session2 = self.conn.open_session()
+        # Use a free_space_target that is guaranteed to run on the small file.
+        session2.compact(uri_small,f'free_space_target={small_file_free_space - 1}MB')
+
+        time.sleep(1)
+        
+        # Periodically check that foreground compaction has done some work on the small table.
+        pages_rewritten = self.get_pages_rewritten(uri_small)
+        while (pages_rewritten == 0):
+            time.sleep(1)
+            pages_rewritten = self.get_pages_rewritten(uri_small)
+                    
+        # Disable the background compaction server.
+        self.session.compact(None,'background=false')
+        
+        # Check that the background server is no longer running.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        self.assertEqual(compact_running, 0)
+        stat_cursor.close()
+    
+    # Test the background server while creating and dropping tables.
+    def test_background_compact_table_ops(self):
+        # Trigger the background compaction server first, before creating tables.
+        self.session.compact(None,'background=true,free_space_target=1MB')
+
+        # Create n tables for background compaction to loop through.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+                
+        # Delete some data from the half the tables.
+        for i in range(self.n_tables):
+            if (i % 2 == 0):
+                self.delete_range(f'{self.uri_prefix}_{i}')
+                
+        # Periodically check that background compaction has done some work on the first table.
```

Comment:
Should we check this is true for at least of the tables?

Reply:
The intent is to test the background compaction while dropping the tables so I just wanted to ensure it's started doing some work before we start dropping tables.

Reply:
Is it possible to drop everything before the server starts working?

Reply:
Why are we only checking the first table? Any table could be compacted, should we try to get the stat on any table and as soon as one shows work has been done we can break from the loop?

Reply:
That's true, background compact could be anywhere in the list of tables.

Reply:
Fixed in https://github.com/wiredtiger/wiredtiger/pull/9487/commits/fda575a199e7a136740e6e440a73fe871ba6ed9a